parsl/data_provider/files.py

```diff
@@ -49,6 +49,7 @@ def __init__(self, url, dman=None):
         self.filename = os.path.basename(self.path)
         self.dman = dman if dman else DataManager.get_data_manager()
         self.data_future = {}
+        self.is_a_parsl_file = True
```

Comment:
A few new things in this commit, we are working to remove any dependencies on outside libraries, including parsl, from the work queue worker since we do not want to assume that there is every machine has parsl installed on it.  Since in the runner we do not have access to the file, `isinstance(file_object, File)` does not work as expected on the runner.  This attribute is added so we can instead do `hasattr(file_object, "is_a_parsl_file)` in the easiest way.  We can add to this check by looking for other attributes to make sure that the object we are looking at is in fact a parsl file object. @nhazekam

Reply:
I'm fairly surprised that `File` deserialises at all if the parsl code isn't available - what kind of object do you get for file_object? (eg, what does `type(file_object)` return at the point you call hasattr(file_object, "is_a_parsl_file" ?)

Reply:
It returns <class parsl.data_provider.files.File>. From what we have found and read, pickling does store the properties of a class.

Reply:
so can you check the type that way rather than adding stuff into the `File` instances?

Reply:
We did try a couple different methods to check against the class.  But when we tried to check against that class with the full name, it required parsl to be present.  We also tried to send the file implementation with the worker script, the class was not resolved in the same way since we didn’t have the rest of parsl present.  We could check against the string returned by `type(file_object)` and just look for the name `parsl.data_provider.files.File` in the type name instead.